backend/errors/error_source.go

```diff
@@ -0,0 +1,37 @@
+package errors
+
+import (
+	"net/http"
+)
+
+func GetErrorSource(statusCode int) ErrorSource {
+	switch statusCode {
+	case http.StatusBadRequest,
+		http.StatusMethodNotAllowed,
+		http.StatusNotAcceptable,
+		http.StatusPreconditionFailed,
+		http.StatusRequestEntityTooLarge,
+		http.StatusRequestHeaderFieldsTooLarge,
+		http.StatusRequestURITooLong,
+		http.StatusExpectationFailed,
+		http.StatusUpgradeRequired,
+		http.StatusRequestedRangeNotSatisfiable,
+		http.StatusNotImplemented:
+		return ErrorSourcePlugin
+	}
+
+	if statusCode < 400 {
+		return ErrorSourceNone
+	}
+
+	return ErrorSourceDownstream
+}
+
+// ErrorSource type defines the source of the error
+type ErrorSource string
+
+const (
+	ErrorSourcePlugin     ErrorSource = "plugin"
+	ErrorSourceDownstream ErrorSource = "downstream"
+	ErrorSourceNone       ErrorSource = "none"
```

Comment:
Nit. Package name is errors so would be enough with `Source`, `SourcePlugin` etc since error is implied from the package name

Reply:
removed